contrib/kubespray/script/openpai-generator.py

```diff
@@ -72,6 +84,59 @@ def generate_template_file(template_file_path, output_path, map_table):
     write_generated_file(output_path, generated_template)
 
 
+def get_kubernetes_node_info_from_API():
+    config.load_kube_config()
+    api_instance = client.CoreV1Api()
+
+    # https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreV1Api.md#list_node
+    pretty = 'true'
+    timeout_seconds = 56
+
+    ret = dict()
+    try:
+        api_response = api_instance.list_node(pretty=pretty, timeout_seconds=timeout_seconds)
+        for node in api_response.items:
+            ret[node.metadata.name] = {
+                "allocatable-cpu-resource": int(parse_quantity(node.status.allocatable['cpu'])),
+                "allocatable-mem-resource": int(parse_quantity(node.status.allocatable['memory']) / 1024 / 1024 / 1024),
+                "allocatable-gpu-resource": int(parse_quantity(node.status.allocatable['nvidia.com/gpu'])),
+            }
+    except ApiException as e:
+        logger.error("Exception when calling CoreV1Api->list_node: %s\n" % e)
+
+    return ret
+
+
+def hived_config_prepare(worker_dict, node_resource_dict):
+    hived_config = dict()
+    hived_config["nodelist"] = []
+
+    min_mem = 100000000
+    min_gpu = 100000000
+    min_cpu = 100000000
+
+    for key in node_resource_dict:
+        if key not in worker_dict:
+            continue
+        if node_resource_dict[key]["allocatable-gpu-resource"] == 0:
+            logger.error("Allocatable GPU number in {0} is 0, Hived doesn't support worker node with 0 GPU".format(key))
+            logger.error("Please remove {0} from your worklist".format(key))
```

Comment:
Please remove {0} from your worklist, or check if the NVIDIA device plugin is running healthy on the node.

Reply:
done